typed-racket-lib/typed-racket/tc-setup.rkt

```diff
@@ -17,9 +17,18 @@
          tc-module/full
          tc-toplevel/full)
 
+(define (authorized-code-inspector?)
+  ;; Check whether the current code inspector has sufficient
+  ;; privileges to access unsafe operations.
+  (with-handlers ([exn:fail? (lambda (e) #f)])
+    (module->namespace 'racket/unsafe/ops)
+    #t))
+
 (define (maybe-optimize body)
   ;; do we optimize?
-  (if (and (optimize?) (not (getenv "PLT_TR_NO_OPTIMIZE")))
```

Comment:
I think we should keep the environment variable -- it might be useful for other things.